stout/http.h

```diff
@@ -0,0 +1,529 @@
+                    &data),
+                CURLM_OK);
+            CHECK_EQ(
+                curl_multi_setopt(
+                    data.multi.handle(),
+                    CURLMOPT_SOCKETFUNCTION,
+                    socket_function),
+                CURLM_OK);
+            CHECK_EQ(
+                curl_multi_setopt(
+                    data.multi.handle(),
+                    CURLMOPT_TIMERDATA,
+                    &data),
+                CURLM_OK);
+            CHECK_EQ(
+                curl_multi_setopt(
+                    data.multi.handle(),
+                    CURLMOPT_TIMERFUNCTION,
+                    timer_function),
+                CURLM_OK);
+
+            // CURL easy options
+            CHECK_EQ(
+                curl_easy_setopt(
+                    data.easy.handle(),
+                    CURLOPT_URL,
+                    data.url.c_str()),
+                CURLE_OK);
+            CHECK_EQ(
+                curl_easy_setopt(
+                    data.easy.handle(),
+                    CURLOPT_WRITEDATA,
+                    &data.context),
+                CURLE_OK);
+            CHECK_EQ(
+                curl_easy_setopt(
+                    data.easy.handle(),
+                    CURLOPT_WRITEFUNCTION,
+                    detail::EasyHandle::WriteCallback),
+                CURLE_OK);
+            CHECK_EQ(
+                curl_easy_setopt(
+                    data.easy.handle(),
+                    CURLOPT_TIMEOUT_MS,
+                    duration_cast<milliseconds>(data.timeout)),
+                CURLE_OK);
+            CHECK_EQ(
+                curl_easy_setopt(
+                    data.easy.handle(),
+                    CURLOPT_NOSIGNAL,
+                    1),
+                CURLE_OK);
+
+            switch (data.method) {
+              case Method::GET:
+                CHECK_EQ(
+                    curl_easy_setopt(
+                        data.easy.handle(),
+                        CURLOPT_HTTPGET,
+                        1),
+                    CURLE_OK);
+                break;
+              case Method::POST:
+                CHECK_EQ(
+                    curl_easy_setopt(
+                        data.easy.handle(),
+                        CURLOPT_HTTPPOST,
+                        1),
+                    CURLE_OK);
+                CHECK_EQ(
+                    curl_easy_setopt(
+                        data.easy.handle(),
+                        CURLOPT_POSTFIELDS,
+                        data.fields_string.c_str()),
+                    CURLE_OK);
+                break;
+            }
+
+            // Start handling connection
+            CHECK_EQ(
+                curl_multi_add_handle(
+                    data.multi.handle(),
+                    data.easy.handle()),
+                CURLM_OK);
+          })
+          .interrupt([](auto& data, auto& k) {
+            data.context.should_interrupt
+                .store(true, std::memory_order_relaxed);
```

Comment:
How can you be sure that it hasn't already completed in which case you'd be calling `k.Start()` and `k.Stop()` here as well?